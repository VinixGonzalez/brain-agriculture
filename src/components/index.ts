export * from "./sidebar/Sidebar";
export * from "./footer/Footer";
export * from "./header/Header";
export * from "./register-producer-form/RegisterProducerForm";
export * from "./input/Input";
