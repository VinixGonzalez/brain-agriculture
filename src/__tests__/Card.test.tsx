import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { Card } from "@/components";

describe("Card component", () => {
  it("renders with default props", () => {
    const { container } = render(<Card />);
    expect(container).toMatchSnapshot();
  });

  it("renders with title and titleIcon", () => {
    const title = "Test Title";
    const titleIcon = <div>Icon</div>;
    const { container } = render(<Card title={title} titleIcon={titleIcon} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with children", () => {
    const children = <div>Children</div>;
    const { container } = render(<Card>{children}</Card>);
    expect(container).toMatchSnapshot();
  });

  it("renders with className", () => {
    const className = "custom-class";
    const { container } = render(<Card className={className} />);
    expect(container).toMatchSnapshot();
  });

  it("renders with title and children", () => {
    const title = "Test Title";
    const children = <div>Children</div>;
    const { container } = render(<Card title={title}>{children}</Card>);
    expect(container).toMatchSnapshot();
  });

  it("renders with titleIcon and children", () => {
    const titleIcon = <div>Icon</div>;
    const children = <div>Children</div>;
    const { container } = render(<Card titleIcon={titleIcon}>{children}</Card>);
    expect(container).toMatchSnapshot();
  });

  it("renders with all props", () => {
    const title = "Test Title";
    const titleIcon = <div>Icon</div>;
    const children = <div>Children</div>;
    const className = "custom-class";
    const { container } = render(
      <Card title={title} titleIcon={titleIcon} className={className}>
        {children}
      </Card>
    );
    expect(container).toMatchSnapshot();
  });
});
