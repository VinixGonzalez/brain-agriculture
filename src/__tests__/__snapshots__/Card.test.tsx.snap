// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders with all props 1`] = `
<div>
  <div
    class="shadow-lg border border-gray-200 p-4 rounded-lg bg-white custom-class"
  >
    <div
      class="flex items-center gap-2 "
    >
      <div>
        <div>
          Icon
        </div>
      </div>
      <p
        class="text-xl"
      >
        Test Title
      </p>
    </div>
    <hr
      class="w-full"
    />
    <div>
      Children
    </div>
  </div>
</div>
`;

exports[`Card component renders with children 1`] = `
<div>
  <div
    class="shadow-lg border border-gray-200 p-4 rounded-lg bg-white undefined"
  >
    <div>
      Children
    </div>
  </div>
</div>
`;

exports[`Card component renders with className 1`] = `
<div>
  <div
    class="shadow-lg border border-gray-200 p-4 rounded-lg bg-white custom-class"
  />
</div>
`;

exports[`Card component renders with default props 1`] = `
<div>
  <div
    class="shadow-lg border border-gray-200 p-4 rounded-lg bg-white undefined"
  />
</div>
`;

exports[`Card component renders with title and children 1`] = `
<div>
  <div
    class="shadow-lg border border-gray-200 p-4 rounded-lg bg-white undefined"
  >
    <div
      class="flex items-center gap-2 "
    >
      <p
        class="text-xl"
      >
        Test Title
      </p>
    </div>
    <hr
      class="w-full"
    />
    <div>
      Children
    </div>
  </div>
</div>
`;

exports[`Card component renders with title and titleIcon 1`] = `
<div>
  <div
    class="shadow-lg border border-gray-200 p-4 rounded-lg bg-white undefined"
  >
    <div
      class="flex items-center gap-2 "
    >
      <div>
        <div>
          Icon
        </div>
      </div>
      <p
        class="text-xl"
      >
        Test Title
      </p>
    </div>
    <hr
      class="w-full"
    />
  </div>
</div>
`;

exports[`Card component renders with titleIcon and children 1`] = `
<div>
  <div
    class="shadow-lg border border-gray-200 p-4 rounded-lg bg-white undefined"
  >
    <div>
      Children
    </div>
  </div>
</div>
`;
